keel-core/src/main/kotlin/com/netflix/spinnaker/keel/actuation/CheckScheduler.kt

```diff
@@ -79,7 +80,10 @@ class CheckScheduler(
                    * to prevent the cancellation of all coroutines under [job]
                    */
                   withTimeout(checkTimeout.toMillis()) {
-                    launch { resourceActuator.checkResource(it) }
+                    launch {
+                      resourceActuator.checkResource(it)
+                      publisher.publishEvent(ResourceCheckCompleted)
```

Comment:
I created a new event that gets published after checkResource completes successfully.

Reply:
I think that because this is in a coroutine (`launch {` starts the coroutine in the background) this will publish right when the check starts. It would be worth verifying.

Reply:
@emjburns The `publishEvent` is called after the `checkResource` method returns.

Reply:
ahh 🤦‍♂ great.